universal-login-example/src/services/StorageService.js

```diff
@@ -0,0 +1,72 @@
+import localforage from 'localforage';
+
+const IDENTITY_KEY = 'universal-login-identity';
+const DEVICES_KEY = 'universal-login-devices';
+
+class StorageService {
+  checkForIdentity() {
+    return localforage.getItem(IDENTITY_KEY).then(function (value) {
+      return value;
+    }).catch(function() {
+      return false;
+    });
+  }
+
+  storeIdentity(identity) {
+    return localforage.setItem(IDENTITY_KEY, identity).then(function () {
+      return true;
+    }).catch(function() {
+      return false;
+    });
+  }
+
+  getDevices() {
+    return localforage.getItem(DEVICES_KEY).then(function (value) {
+      return value;
+    }).catch(function() {
+      return false;
```

Comment:
Should return an empty array. Functions should always return the same type.
If there are items - an array of items if no items - empty array.

Same is true for other functions. Should return null, undefined or throw an exception.

Returning different types makes it harder to use, test and destroys JIT compiler optimisations.